tests/query_runner/test_basesql_queryrunner.py

```diff
@@ -0,0 +1,74 @@
+import unittest
+
+from redash.query_runner import BaseSQLQueryRunner, BaseQueryRunner
+
+
+class TestBaseSQLQueryRunner(unittest.TestCase):
+
+    def setUp(self):
+        self.query_runner = BaseSQLQueryRunner(BaseQueryRunner)
+
+    def test_apply_auto_limit_origin_no_limit_1(self):
+        origin_query_text = "SELECT 2"
+        query_text = self.query_runner.apply_auto_limit(origin_query_text)
+        self.assertEqual("SELECT 2 LIMIT 1000", query_text)
+
+    def test_apply_auto_limit_origin_have_limit_1(self):
+        origin_query_text = "SELECT 2 LIMIT 100"
+        query_text = self.query_runner.apply_auto_limit(origin_query_text)
+        self.assertEqual(origin_query_text, query_text)
+
+    def test_apply_auto_limit_origin_have_limit_2(self):
+        origin_query_text = "SELECT * FROM fake WHERE id IN (SELECT id FROM fake_2 LIMIT 200) LIMIT 200"
+        query_text = self.query_runner.apply_auto_limit(origin_query_text)
+        self.assertEqual(origin_query_text, query_text)
+
+    def test_apply_auto_limit_origin_no_limit_2(self):
+        origin_query_text = "SELECT * FROM fake WHERE id IN (SELECT id FROM fake_2 LIMIT 200)"
+        query_text = self.query_runner.apply_auto_limit(origin_query_text)
+        self.assertEqual(origin_query_text + " LIMIT 1000", query_text)
+
+    def test_apply_auto_limit_non_select_query(self):
+        origin_query_text = "create table execution_times as " \
+                            "(select id, retrieved_at, data_source_id, query, runtime, query_hash " \
+                            "from query_results order by 1 desc)"
```

Comment:
Instead of using backslashes on every line when writing long string, it's better form to use implicit line continuation inside of braces here, e.g. 
```suggestion
        origin_query_text = (
            "create table execution_times as "
            "(select id, retrieved_at, data_source_id, query, runtime, query_hash "
            "from query_results order by 1 desc)"
        )
```
or triple quotes:
```suggestion
        origin_query_text = """
            create table execution_times as
            (select id, retrieved_at, data_source_id, query, runtime, query_hash
            from query_results order by 1 desc)
        """
```